src/image/utils.js

```diff
@@ -112,29 +112,28 @@ function isImageAllowedInParent( selection, schema, model ) {
 	return schema.checkChild( parent, 'image' );
 }
 
-// Check used in image commands for additional cases when the command should be disabled:
-//
-// - selection is on object
-// - selection is inside object
+// Check if selection is on object.
 //
 // @returns {Boolean}
-function checkSelectionWithObject( selection, schema ) {
+function checkSelectionOnObject( selection, schema ) {
 	const selectedElement = selection.getSelectedElement();
 
-	const isSelectionOnObject = !!selectedElement && schema.isObject( selectedElement );
-	const isSelectionInObject = !![ ...selection.focus.getAncestors() ].find( ancestor => schema.isObject( ancestor ) );
+	return !( !!selectedElement && schema.isObject( selectedElement ) );
```

Comment:
Doesn't this return the opposite thing? If this is checking if the selection is on object it should simply return:

`return selectedElement && schema.isObject( selectedElement );`